src/helper/alt-ips-client.ts

```diff
@@ -0,0 +1,122 @@
+import os from 'node:os';
+import config from 'config';
+import is from '@sindresorhus/is';
+import _ from 'lodash';
+import { scopedLogger } from '../lib/logger.js';
+import got, { RequestError } from 'got';
+import type { Socket } from 'socket.io-client';
+import { pluralize } from '../lib/util.js';
+
+const mainLogger = scopedLogger('general');
+const altIpsLogger = scopedLogger('api:connect:alt-ips-handler');
+
+export class AltIpsClient {
+	private readonly INTERVAL_TIME = 10 * 60 * 1000;
+	private timer?: NodeJS.Timeout;
+	private socket: Socket;
+	private ip: string;
+	private currentIps: string[] = [];
+	private currentRejectedIps: string[] = [];
+
+	constructor (socket: Socket, ip: string) {
+		this.socket = socket;
+		this.ip = ip;
+	}
+
+	updateConfig (socket: Socket, ip: string) {
+		this.socket = socket;
+		this.ip = ip;
+		this.start();
+	}
+
+	start () {
+		clearTimeout(this.timer);
+		this.run();
+	}
+
+	private run () {
+		void this.refreshAltIps().catch((error) => {
+			altIpsLogger.error(error);
+		}).finally(() => {
+			this.timer = setTimeout(() => this.run(), this.INTERVAL_TIME);
+		});
+	}
+
+
+	async refreshAltIps (): Promise<void> {
+		const addresses = _(os.networkInterfaces())
+			.values()
+			.filter(is.truthy)
+			.flatten()
+			.uniqBy('address')
+			.filter(address => !address.internal)
+			.filter(address => !address.address.startsWith('fe80:')) // filter out link-local addresses
+			.filter(address => !address.address.startsWith('169.254.')) // filter out link-local addresses
+			.value();
+
+		const results = await Promise.allSettled(addresses.map(({ address, family }) => this.getAltIpToken(address, family === 'IPv6' ? 6 : 4)));
+
+		const ipsToTokens: [string, string][] = [];
+		const rejectedLocalIpsToReasons: Record<string, string> = {};
+		results.forEach((result, index) => {
+			if (result.status === 'fulfilled') {
+				ipsToTokens.push([ result.value.ip, result.value.token ]);
+			} else {
+				if (result.reason instanceof RequestError) {
+					rejectedLocalIpsToReasons[`${addresses[index]!.address} (local)`] = result.reason.message;
+				} else {
+					rejectedLocalIpsToReasons[`${addresses[index]!.address} (local)`] = (result.reason as Error).toString();
+				}
```

Comment:
The original logic here was better. Differentiate between "rejected by API" and "failed for unknown reason". Also, keep the original message formatting (one IP per log call if failed).